### DISCUSSION TITLE
Is there anything similar to the ActualWidth and ActualHeight properties in WPF in Avalonia?

### DISCUSSION THREAD

**Discussion by qqJoker** [AvaloniaUI/Avalonia#12973] [Q&A] [answered]
When I tried to use ScaleTransform to scale the control, I found that the control did not have any properties to represent the currently rendered size. The size displayed by Bounds was always fixed. I did not find anything similar to ActualWidth or ActualHeight in WPF in Avalonia. property, so now I can only get the visual width through (Bounds.Width * ScaleTransform.ScaleX)?

**Comment by timunie** [comment]
`this.GetTransformedBounds();` should work. Add `using Avalonia.VisualTree;`

**Comment by maxkatz6** [comment, marked as answer]
`ActualWidth+ActualHeight` can be replaced with `Bounds.Size` property.

`GetTransformedBounds` is also useful when you want to get transformed value, but that's not exactly what wpf properties return.

**Reply by qqJoker** [reply]
thanks so much
